lib/argon2.js

```diff
@@ -58,63 +54,135 @@
      *      .then(h => console.log(h.hash, h.hashHex, h.encoded))
      *      .catch(e => console.error(e.message, e.code))
      */
+
     function argon2Hash(params) {
-        if (!scriptLoadedPromise) {
-            scriptLoadedPromise = loadScript(params.asmFileUrl);
+        if (!global.WebAssembly) {
+            return new Promise(function (resolve, reject) {
+                loadScript(params.asmFileUrl)
+                    .then(function () {
+                        resolve(calcHash(params));
+                    })
+                    .catch(function (error) {
+                        reject("Error loading argon2-asm.min.js: " + error + ", " + params.cacheKey);
```

Comment:
From your own documentation above, `params.cacheKey` doesn't seem to exist. Is it really something this library is supposed to be aware of?